internal/backend/usageupdater/usagedata.go

```diff
@@ -0,0 +1,105 @@
+package usageupdater
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+	"go.autokitteh.dev/autokitteh/internal/version"
+	"go.autokitteh.dev/autokitteh/internal/xdg"
+	"go.uber.org/zap"
+)
+
+type updateRequest struct {
+	InstallationID string `json:"installationID"`
+	Version        string `json:"version"`
+	Commit         string `json:"commit"`
+	OS             string `json:"os"`
+	Arch           string `json:"arch"`
+}
+
+type UsageUpdater interface {
+	Start()
+	Stop()
+}
+
+type usageUpdater struct {
+	installationID uuid.UUID
+	updateInterval time.Duration
+	shutdownChan   chan struct{}
+	poster         poster
+}
+
+func New(z *zap.Logger, cfg *Config) (UsageUpdater, error) {
+	if !cfg.Enabled {
+		return &nopUpdater{}, nil
+	}
+
+	installationIDFile := filepath.Join(xdg.ConfigHomeDir(), "installID")
+
+	if _, err := os.Stat(installationIDFile); os.IsNotExist(err) {
+		id := uuid.New()
+		if err := os.WriteFile(installationIDFile, []byte(id.String()), 0600); err != nil {
+			return nil, err
+		}
+	}
+
+	data, err := os.ReadFile(installationIDFile)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := uuid.Parse(string(data))
```

Comment:
if this fails we'll need to recreate it as well?

Reply:
good point, 
tried to resolve it
make the code a bit more "error handli"
let me know what you think